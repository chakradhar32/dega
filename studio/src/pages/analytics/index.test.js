import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { useDispatch, Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';

import '../../matchMedia.mock';
import Analytics from './index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

let state = {
  spaces: {
    orgs: [{ id: 1, organazation: 'Organization 1', spaces: [11] }],
    details: {
      11: {
        id: 11,
        name: 'Space 11',
        analytics: {
          plausible: {
            domain: 'domain',
            embed_code: 'embed-code',
            server_url: 'url',
          },
        },
      },
    },
    loading: false,
    selected: 11,
  },
};

describe('Analytics Component', () => {
  let store;
  let mockedDispatch;

  beforeEach(() => {
    store = mockStore({});
    store.dispatch = jest.fn(() => ({}));
    mockedDispatch = jest.fn();
    useDispatch.mockReturnValue(mockedDispatch);
  });
  describe('snapshot testing', () => {
    it('should render the component', () => {
      store = mockStore(state);
      const tree = mount(
        <Provider store={store}>
          <Router>
            <Analytics />
          </Router>
        </Provider>,
      );
      expect(tree).toMatchSnapshot();
    });
    it('should render the component when loading', () => {
      const state2 = { ...state };
      state2.spaces.loading = true;
      const store2 = mockStore(state2);
      const tree = mount(
        <Provider store={store2}>
          <Router>
            <Analytics />
          </Router>
        </Provider>,
      );
      expect(tree).toMatchSnapshot();
    });
  });

  it('should render the component when no embed_code found', () => {
    const state2 = { ...state };
    state2.spaces = {
      orgs: [{ id: 1, organazation: 'Organization 1', spaces: [11] }],
      details: {
        11: {
          id: 11,
          name: 'Space 11',
          analytics: {
            plausible: {
              domain: 'domain',
              server_url: 'url',
            },
          },
        },
      },
      loading: false,
      selected: 11,
    };
    const store2 = mockStore(state2);
    const tree = mount(
      <Provider store={store2}>
        <Router>
          <Analytics />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
