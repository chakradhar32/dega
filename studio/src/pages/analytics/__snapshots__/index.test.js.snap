// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics Component should render the component when no embed_code found 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Analytics>
        <Space
          direction="vertical"
        >
          <div
            className="ant-space ant-space-vertical"
            style={Object {}}
          >
            <Item
              className="ant-space-item"
              direction="vertical"
              index={0}
              key="ant-space-item-0"
              marginDirection="marginRight"
              wrap={false}
            >
              <div
                className="ant-space-item"
                style={Object {}}
              >
                <Result
                  title="No analytics found"
                >
                  <div
                    className="ant-result ant-result-info"
                  >
                    <div
                      className="ant-result-icon"
                    >
                      <ForwardRef(ExclamationCircleFilled)>
                        <AntdIcon
                          icon={
                            Object {
                              "icon": Object {
                                "attrs": Object {
                                  "focusable": "false",
                                  "viewBox": "64 64 896 896",
                                },
                                "children": Array [
                                  Object {
                                    "attrs": Object {
                                      "d": "M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 010-96 48.01 48.01 0 010 96z",
                                    },
                                    "tag": "path",
                                  },
                                ],
                                "tag": "svg",
                              },
                              "name": "exclamation-circle",
                              "theme": "filled",
                            }
                          }
                        >
                          <span
                            aria-label="exclamation-circle"
                            className="anticon anticon-exclamation-circle"
                            role="img"
                          >
                            <IconReact
                              icon={
                                Object {
                                  "icon": Object {
                                    "attrs": Object {
                                      "focusable": "false",
                                      "viewBox": "64 64 896 896",
                                    },
                                    "children": Array [
                                      Object {
                                        "attrs": Object {
                                          "d": "M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 010-96 48.01 48.01 0 010 96z",
                                        },
                                        "tag": "path",
                                      },
                                    ],
                                    "tag": "svg",
                                  },
                                  "name": "exclamation-circle",
                                  "theme": "filled",
                                }
                              }
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="exclamation-circle"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                key="svg-exclamation-circle"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
                                  key="svg-exclamation-circle-svg-0"
                                />
                              </svg>
                            </IconReact>
                          </span>
                        </AntdIcon>
                      </ForwardRef(ExclamationCircleFilled)>
                    </div>
                    <div
                      className="ant-result-title"
                    >
                      No analytics found
                    </div>
                  </div>
                </Result>
              </div>
            </Item>
          </div>
        </Space>
      </Analytics>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Analytics Component snapshot testing should render the component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Analytics>
        <Space
          direction="vertical"
        >
          <div
            className="ant-space ant-space-vertical"
            style={Object {}}
          >
            <Item
              className="ant-space-item"
              direction="vertical"
              index={0}
              key="ant-space-item-0"
              marginDirection="marginRight"
              wrap={false}
            >
              <div
                className="ant-space-item"
                style={Object {}}
              >
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "embed-code",
                    }
                  }
                />
              </div>
            </Item>
          </div>
        </Space>
      </Analytics>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Analytics Component snapshot testing should render the component when loading 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Analytics>
        <Skeleton
          avatar={false}
          paragraph={true}
          title={true}
        >
          <div
            className="ant-skeleton"
          >
            <div
              className="ant-skeleton-content"
            >
              <Title
                prefixCls="ant-skeleton-title"
                width="38%"
              >
                <h3
                  className="ant-skeleton-title"
                  style={
                    Object {
                      "width": "38%",
                    }
                  }
                />
              </Title>
              <Paragraph
                prefixCls="ant-skeleton-paragraph"
                rows={3}
                width="61%"
              >
                <ul
                  className="ant-skeleton-paragraph"
                >
                  <li
                    key="0"
                    style={
                      Object {
                        "width": undefined,
                      }
                    }
                  />
                  <li
                    key="1"
                    style={
                      Object {
                        "width": undefined,
                      }
                    }
                  />
                  <li
                    key="2"
                    style={
                      Object {
                        "width": "61%",
                      }
                    }
                  />
                </ul>
              </Paragraph>
            </div>
          </div>
        </Skeleton>
      </Analytics>
    </Router>
  </BrowserRouter>
</Provider>
`;
